src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -707,6 +742,30 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
             }
         }
 
+        private BoundExpression BindDiscardedExpression(
+            DiscardedDesignationSyntax designation,
+            TypeSyntax typeSyntax,
+            DiagnosticBag diagnostics)
+        {
+            bool hasErrors = false;
+            bool isVar;
+            bool isConst = false;
+            AliasSymbol alias;
+            TypeSymbol declType = BindVariableType(designation, diagnostics, typeSyntax, ref isConst, out isVar, out alias);
+
+            if (!isVar)
+            {
+                if (designation.Parent.Kind() == SyntaxKind.ParenthesizedVariableDesignation)
+                {
+                    // An explicit type can only be provided next to the wildcard
+                    Error(diagnostics, ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, designation);
+                    hasErrors = true;
+                }
+            }
```

Comment:
I'd like to move this diagnostic into the caller, so the type is bound once and this error is reported once.